### PR TITLE
Comments for VertexData DO NOT MERGE

### PR DESCRIPTION
**Pull request by BabylonJSGuide**
Do not merge this PR as it is not completed. Having looked through files in 'src' to add comments to I have had difficulty because I am not able to determine what many of the entities do and so cannot add comments. I thought I would be OK with the VertexData file but have a number of questions already and am not sure whether to or how to proceed.

Questions 
How to describe uvs2 to uvs6 in at the start of class VertexData?

I have made a start with CreateRibbon and CreateBox but am I adding too much explanation at this point or should this detail be left for the methods of the MeshBuilder class that call VertexData?

Do the other comments I have written make sense?

My apologies but even when I finish VertexData I have come to the point where I cannot help any further with adding comments.

### CODE REVIEW

File: src/Mesh/babylon.mesh.vertexData.ts [hunk 1]
```diff
@@ -8,23 +11,50 @@
...
+         * A linear array of the the x, y, z normal vector of each vertex  [...., x, y, z, .....]
+         */
         public normals: Nullable<FloatArray>;
+        /**
+         * A linear array of the the x, y, z tangent vector of each vertex  [...., x, y, z, .....]
+         */
         public tangents: Nullable<FloatArray>;
+        /**
+         * A linear array of the the u,v mapping of the texture image of each vertex  [...., u, v, .....]
+         */
```
Comment (deltakosh) [inline src/Mesh/babylon.mesh.vertexData.ts]: Meshes can have multiple set of texture coordinates. For instance diffuse texture can be on set 1 and bump on set 5

File: src/Mesh/babylon.mesh.vertexData.ts [hunk 2]
```diff
@@ -8,23 +11,50 @@
...
+         * A linear array of the r, g, b, color of each vertex  [...., r, g, b, a, .....]
+         */
         public colors: Nullable<FloatArray>;
         public matricesIndices: Nullable<FloatArray>;
         public matricesWeights: Nullable<FloatArray>;
         public matricesIndicesExtra: Nullable<FloatArray>;
         public matricesWeightsExtra: Nullable<FloatArray>;
+        /**
+         * A linear array of the i, j, k indices for each triangular facet  [...., i, j, k .....]
+         * where index i gives positions[3 * i],  gives positions[3 * i + 1], gives positions[3 * i + 2]
```
Comment (deltakosh) [inline src/Mesh/babylon.mesh.vertexData.ts]: Probably me but this sentence looks weird

File: src/Mesh/babylon.mesh.vertexData.ts [hunk 3]
```diff
@@ -8,23 +11,50 @@
         setIndices(indices: IndicesArray, totalVertices: Nullable<number>, updatable?: boolean): void;
     }
 
+    /**
+     * This class contains the various kinds of data on every vertex of a mesh used in determining its shape and appearance 
+     */
     export class VertexData {
+        /**
+         * A linear array of the the x, y, z position of each vertex  [...., x, y, z, .....]
```
Comment (deltakosh) [inline src/Mesh/babylon.mesh.vertexData.ts]: Why linear ?

File: src/Mesh/babylon.mesh.vertexData.ts [hunk 4]
```diff
@@ -8,23 +11,50 @@
...
         public uvs6: Nullable<FloatArray>;
+        /**
+         * A linear array of the r, g, b, color of each vertex  [...., r, g, b, a, .....]
+         */
         public colors: Nullable<FloatArray>;
         public matricesIndices: Nullable<FloatArray>;
         public matricesWeights: Nullable<FloatArray>;
         public matricesIndicesExtra: Nullable<FloatArray>;
         public matricesWeightsExtra: Nullable<FloatArray>;
+        /**
```
Comment (deltakosh) [inline src/Mesh/babylon.mesh.vertexData.ts]: Matrices indices is the index in the array of matrices produced by bones. Each vertex have up to 4 indices (8 if the mstricesIndicesExtra is set).
Weights defines the weight of each indexed matrix in the final computation

File: src/Mesh/babylon.mesh.vertexData.ts [hunk 5]
```diff
@@ -74,8 +104,10 @@
 
         /**
          * Associates the vertexData to the passed Mesh.  
-         * Sets it as updatable or not (default `false`).  
-         * Returns the VertexData.  
+         * Sets it as updatable or not (default `false`) 
+         * @param mesh the mesh the vertexData is applied to 
+         * @param updatable when used and having the value true allows new data to update the vertexData 
+         * @returns the VertexData.
```
Comment (deltakosh) [inline src/Mesh/babylon.mesh.vertexData.ts]: No period at the end

File: src/Mesh/babylon.mesh.vertexData.ts [hunk 6]
```diff
@@ -288,8 +329,9 @@
         }
 
         /**
-         * Merges the passed VertexData into the current one.  
-         * Returns the modified VertexData.  
+         * Merges the passed VertexData into the current one
+         * @param other the VetexData to be merged into the current one
```
Comment (deltakosh) [inline src/Mesh/babylon.mesh.vertexData.ts]: Typo

File: src/Mesh/babylon.mesh.vertexData.ts [hunk 7]
```diff
@@ -797,8 +859,19 @@
         }
 
         /**
-         * Creates the VertexData of the Box.  
-         */
+         * Creates the VertexData for a box
+         * @param options an object used to set the following optional parameters for the box, required but can be empty
```
Comment (deltakosh) [inline src/Mesh/babylon.mesh.vertexData.ts]: You can check what is in MeshBuilder for these guys:)

**Comment by BabylonJSGuide** [comment]
Have updated the comments in babylon.mesh.vertexData.ts and it needs reviewing. However as you can probably see I have had many problems. When I pull Babylon.js I keep getting conflicts  from dist/preview/babylon.d.ts and I do not know a simple way of keeping the pulled babylon.d.ts file and overwriting the local one. Also when I try gulp typedoc-check I get this

`User@User-PC MINGW32 ~/Documents/GitHub/Babylon.js/Tools/Gulp (master)
$ gulp -v
[17:00:50] CLI version 3.9.1
[17:00:50] Local version 3.9.1

User@User-PC MINGW32 ~/Documents/GitHub/Babylon.js/Tools/Gulp (master)
$ gulp typedoc-check
module.js:538
    throw err;
    ^

Error: Cannot find module 'dts-bundle'
    at Function.Module._resolveFilename (module.js:536:15)
    at Function.Module._load (module.js:466:25)
    at Module.require (module.js:579:17)
    at require (internal/module.js:11:18)
    at Object.<anonymous> (C:\Users\User\Documents\GitHub\Babylon.js\Tools\Gulp\gulpfile.js:33:17)
    at Module._compile (module.js:635:30)
    at Object.Module._extensions..js (module.js:646:10)
    at Module.load (module.js:554:32)
    at tryModuleLoad (module.js:497:12)
    at Function.Module._load (module.js:489:3)

User@User-PC MINGW32 ~/Documents/GitHub/Babylon.js/Tools/Gulp (master)
$
`

**Comment by deltakosh** [comment]
No worry. I  suggest restarting a fresh one:)

**Comment by BabylonJSGuide** [comment]
Will start again with a fresh fork.

**Pull request closed without merging**